packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,439 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import {
+  ASSET_NAMESPACE,
+  CHAIN_NAMESPACE,
+  CHAIN_REFERENCE,
+  fromAssetId,
+  solAssetId,
+  toAssetId,
+  wrappedSolAssetId,
+} from '@shapeshiftoss/caip'
+import type { GetFeeDataInput } from '@shapeshiftoss/chain-adapters'
+import type { KnownChainIds } from '@shapeshiftoss/types'
+import { bnOrZero, convertDecimalPercentageToBasisPoints } from '@shapeshiftoss/utils'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { PublicKey, type TransactionInstruction } from '@solana/web3.js'
+import type { AxiosError } from 'axios'
+import { v4 as uuid } from 'uuid'
+
+import { getDefaultSlippageDecimalPercentageForSwapper } from '../../../constants'
+import type {
+  CommonTradeQuoteInput,
+  GetTradeRateInput,
+  ProtocolFee,
+  SwapErrorRight,
+  SwapperDeps,
+  TradeQuote,
+  TradeRate,
+} from '../../../types'
+import { SwapperName, TradeQuoteError } from '../../../types'
+import { getRate, makeSwapErrorRight } from '../../../utils'
+import type { Instruction } from '../models/Instruction'
+import type { QuoteResponse } from '../models/QuoteResponse'
+import { getJupiterQuote, getJupiterSwapInstructions, isSupportedChainId } from '../utils/helpers'
+
+const tradeQuoteMetadata: Map<string, QuoteResponse> = new Map()
+
+export const getTradeRate = async (
```

Comment:
preferably-blocking: extract me into a `getTradeRate.ts` method similar to other swappers

Reply:
Unapplicable anymore, closing